### DISCUSSION TITLE
How can I remove blackhole audio driver from Audio midi setup without deleting it from HAL Folder ?

### DISCUSSION THREAD

**Discussion by VigneshDevanathan15** [ExistentialAudio/BlackHole#707] [General]
Hi community, 
     I can successfully able to install Blackhole audio driver into my Mac. By that I can able to create aggregate device using blackhole and built in microphone.

My question is : 
**I can able to delete aggregated device that created by me while could not able to delete Blackhole driver.
how to destroy it without deleting it from HAL folder and install it again in audio midi without restarting. Any help will be much appreciated**

<img width="784" alt="Screenshot 2023-02-16 at 4 37 13 PM" src="https://user-images.githubusercontent.com/116871440/219350623-cc8e4f68-3557-4fa2-91cd-ec623f310b78.png">

Note: Loom recorder app creating and deleting their  virtual device without even deleting it from HAL folder and restarting.

**Comment by devinroth** [comment]
Are you developing an app?

Devin

**Reply by VigneshDevanathan15** [reply]
Thanks for the reply.

  Yes, actually am developing an app that records audio of the user's system. So I need blackhole to be dynamically shows and destroys itself when starting and stopping the audio recorder respectively. any help will be appreciated. thanks

**Reply by devinroth** [reply]
Yes that can be done with BlackHole. The trick is kAudioBoxPropertyAcquired. Beyond that is getting into my consulting territory. But if you're planning on licensing BlackHole for the app I can give you more info. Contact me at devinroth@existential.audio.

**Reply by VigneshDevanathan15** [reply]
Thanks, 
  I got to know that 'kAudioDevicePropertyIsHidden' property can be used to hide and show the driver.By using this property can we get dynamically show and hide audio driver ? and I checked 'kAudioBoxPropertyAcquired' this  where we can use this to get the property, since I could not get any doc referring this.

**Reply by VigneshDevanathan15** [reply]
Hi,  
I have searched for the docs that referring about this property 'kAudioBoxPropertyAcquired', could not get any. Do I need to license the blackhole to use this property or is it provided inbuilt by Apple ? and is there any licensed version of blackhole driver that can be achieve this and other features or just like same as this open source one ? any update will appreciated thanks.
